app/controllers/posts_controller.rb

```diff
@@ -31,6 +33,20 @@ def create
     end
   end
 
+  def destroy
+    @post = Post.find(params[:id])
+    Comment.where(post_id: @post.id).destroy_all
+    Like.where(post_id: @post.id).destroy_all
+
```

Comment:
- [OPTIONAL] There is a helper that will destroy all dependents of a model. It can be used when defining the relationship between models. 

```ruby
has_many :likes, dependent: :destroy
has_many :comments, dependent: :destroy
``` 

This code above will delete all `comments` and `likes` instances associated with the `Post` model. 👍 😎